src/Compilers/Core/Portable/MetadataReader/MetadataHelpers.cs

```diff
@@ -737,7 +743,7 @@ internal static string UnmangleMetadataNameForArity(string emittedTypeName, int
                     }
 
                     // Account for the dot following THIS namespace name.
-                    if (namespaceNameLength != 0)
+                    if (!isGlobalNamespace)
                     {
                         namespaceNameLength++;
```

Comment:
Why do we increment this on every iteration of the loop?  Given that it seems to track the value passed into the function it seems that it should only be incremented once?

I'm positive that I'm just missing something basic here.